### PR TITLE
chore(Datagrid): add missing stories

### LINKED ISSUE #6074: Update Datagrid stories and docs
```[tasklist]
### Tasks
- [ ] Re-add missing table header story
- [ ] Re-add missing column alignment story
- [ ] Add section on resizable columns to docs
```

### PR DESCRIPTION
**Pull request by devadula-nandan**
Closes #6074 

adds `Column alignment`, `Header` stories

#### What did you change?
packages/ibm-products/src/components/Datagrid/Datagrid.stories.jsx

#### How did you test and verify your work?
storybook

### CODE REVIEW

**Comment by elycheea** [comment]
@devadula-nandan Hmm, that could work. I think it could be more helpful to group the content on resizing columns together though. It might not be as straightforward to find it under column structure. Would love @AlexanderMelox thoughts on this too.

https://github.com/carbon-design-system/carbon-for-products-website/issues/197#issuecomment-2358651644

My suggestion would be to group these two sections (under Column structure) into *Resizable columns* and place it with the auto-size section? Maybe auto-size still needs its own section but having resizable columns next to auto-size could be a way to help adopters understand the different options.

#### Resizable columns
image
image

For headers, a basic variant is enough for now. I believe we previously showed a single header story but also included `useDenseHeader` as one of the options so that we could toggle between the two.

**Review by AlexanderMelox — DISMISSED** [review]

**Review by davidmenendez — DISMISSED** [review]

File: packages/ibm-products/src/components/Datagrid/Datagrid.stories.jsx [hunk 1]
```diff
@@ -311,6 +313,94 @@ export const SelectableRow = () => {
   return <Datagrid datagridState={{ ...datagridState }} />;
 };
 
+export const Header = () => {
+  const [data] = useState(makeData(10));
+  const columns = React.useMemo(() => getColumns(data), []);
```
[the remark this answers is not in the document]
Comment (devadula-nandan) [reply packages/ibm-products/src/components/Datagrid/Datagrid.stories.jsx:318]: interesting

File: packages/ibm-products/src/components/Datagrid/Datagrid.docs-page.js [hunk 2]
```diff
@@ -212,6 +182,40 @@ const App = () => {
           `,
         },
       },
+      {
+        title: 'Resizable columns',
+        description: 'Columns are resizable by default',
```
Comment (elycheea) [inline packages/ibm-products/src/components/Datagrid/Datagrid.docs-page.js:187]:
```suggestion
        description: 'Columns are resizable by default.',
```

**Review by elycheea — DISMISSED** [review]
Just one missed period, but thank you @devadula-nandan!

**Review by elycheea — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Review by AlexanderMelox — APPROVED** [review]

**Pull request merged**